### PR TITLE
feat(api): add email-only notifier for high/medium API errors (#809)

### PR DESCRIPTION
**Pull request by otomatty**
Epic #616 Phase 3 / sub-issue #809。重要 API エラー (severity=high/medium) の
メール通知サービス `services/notifier.ts` を新設し、AI 解析コールバック
(`PUT /api/webhooks/github/ai-result/:id`) の成功時に 1 回だけ呼ぶよう配線する。

Slack はこのリポジトリでは使わないため非対応。`MONITORING_NOTIFY_EMAIL`
未設定 / severity が low/unknown の場合は no-op。本文には sentry_issue_id /
severity / title / 管理画面 URL のみ載せ、Authorization / Cookie / 内部 URL
など PII は混入させない。

Phase 3 / #809: introduce `notifier.ts` for email-only alerts on
high/medium severity API errors. Slack is intentionally unsupported.
The notifier no-ops when `MONITORING_NOTIFY_EMAIL` is unset or severity
is low/unknown. The AI callback is the single call site so duplicate
alerts can't happen, and sends are fire-and-forget so the webhook
response isn't delayed by Resend.

## Summary by CodeRabbit

* **New Features**
  * Email notifications for API errors (medium/high) with optional admin-dashboard link.
  * Notifications are triggered only when severity transitions to a notifiable level, avoiding duplicate alerts on retries or partial updates.

* **Chores**
  * Added MONITORING_NOTIFY_EMAIL and ADMIN_BASE_URL environment settings.

* **Tests**
  * New unit and integration tests covering notification behavior, delivery edge cases, and email content rules.

### CODE REVIEW

File: server/api/src/routes/webhooks/githubAiCallback.ts [hunk 1]
```diff
@@ -161,6 +162,21 @@ app.put("/:id", async (c) => {
...
+    // 重要エラーのメール通知 (Phase 3 / issue #809)。severity が `high` /
+    // `medium` 以外、もしくは `MONITORING_NOTIFY_EMAIL` 未設定の場合は no-op。
+    // ここを唯一の呼び出し元にして二重通知を防ぐ。fire-and-forget で
+    // webhook 応答を遅らせない（notifier 側でエラーは swallow 済み）。
+    //
+    // Email alert for high-impact errors (Phase 3 / #809). The notifier is a
+    // no-op when severity is `low`/`unknown` or `MONITORING_NOTIFY_EMAIL` is
+    // unset. This is the single call site to prevent duplicate alerts; we
+    // fire-and-forget so the webhook response doesn't await Resend.
+    void notifyApiErrorAlert({
```
[the remark this answers is not in the document]
Comment (otomatty) [reply server/api/src/routes/webhooks/githubAiCallback.ts]: Good catch. Fixed in 96da9b6.

The route now reads the row via `getApiErrorById` before `updateAiAnalysis` and gates the notifier on a new helper `severityBecameNotifiable(prev, next)` which returns true only when `prev` was NOT in `{high, medium}` AND `next` IS. This covers the cases you raised:

- **Empty body / partial callback** (e.g. only `ai_summary`): post severity equals pre severity → no notify.
- **Idempotent retry** (`high` → `high`): no transition → no notify.
- **Lateral move** (`medium` → `high`): both notifiable → no notify (chosen so each row produces at most one email; medium→high escalation is intentionally collapsed into the original alert).
- **Downgrade** (`high` → `low`): post is not notifiable → no notify.
- **First-sight escalation** (`unknown`/`low` → `high`/`medium`): exactly one notify.

Tests added in `githubAiCallback.test.ts` cover all four no-notify paths plus the positive transition.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: server/api/.env.example [hunk 2]
```diff
@@ -27,3 +27,22 @@ GITHUB_APP_ID=
...
+#
+# Epic #616 Phase 3（issue #809）用の通知設定。
+# - MONITORING_NOTIFY_EMAIL: severity が `high` / `medium` の API エラー通知の
+#   宛先メール。未設定時は notifier が no-op となり、AI コールバックは正常に
+#   動作する（ステージング段階で通知のみ無効化できる）。
+# - ADMIN_BASE_URL: 管理画面の公開オリジン（例: `https://admin.zedi-note.app`）。
+#   通知メール本文に `<base>/errors/<api_error_id>` のリンクを差し込む際に使う。
+#   内部 URL を設定してはいけない。未設定時はリンク行を省略する。
+MONITORING_NOTIFY_EMAIL=
+ADMIN_BASE_URL=
```
[the remark this answers is not in the document]
Comment (otomatty) [reply server/api/.env.example]: Done in 96da9b6 — `ADMIN_BASE_URL` is now placed above `MONITORING_NOTIFY_EMAIL` in `server/api/.env.example`. The accompanying comment block was kept intact.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: server/api/src/services/notifier.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+ * only summary fields (sentry_issue_id, severity, title) and the public admin
+ * URL — Authorization / Cookie / raw email values are never propagated here.
+ *
+ * @see ./emailService.ts
+ * @see ../routes/webhooks/githubAiCallback.ts
+ * @see https://github.com/otomatty/zedi/issues/616
+ * @see https://github.com/otomatty/zedi/issues/809
+ */
+import type { ApiErrorSeverity } from "../schema/apiErrors.js";
+import { sendEmail } from "./emailService.js";
```
[the remark this answers is not in the document]
Comment (otomatty) [reply server/api/src/services/notifier.ts:34]: Adopted in 96da9b6. `notifier.ts` now imports `getOptionalEnv` from `../lib/env.js` and uses it for both `ADMIN_BASE_URL` and `MONITORING_NOTIFY_EMAIL`, matching the pattern in `emailService.ts`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: server/api/src/services/notifier.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+ * 当サービス内でも参照しない。
+ *
+ * Build subject + HTML body for the alert email. Only sentry_issue_id,
+ * severity, title, and the public admin URL (when `ADMIN_BASE_URL` is set)
+ * are rendered. Authorization, Cookie, raw email, and internal URLs are
+ * never referenced — by contract this function takes only the fields it
+ * renders.
+ */
+function buildEmail(payload: NotifyApiErrorAlertPayload): { subject: string; html: string } {
+  const adminBase = normalizeBaseUrl(process.env.ADMIN_BASE_URL);
```
[the remark this answers is not in the document]
Comment (otomatty) [reply server/api/src/services/notifier.ts]: Applied in 96da9b6:

```ts
const adminBase = normalizeBaseUrl(getOptionalEnv("ADMIN_BASE_URL"));
```

The signature of `normalizeBaseUrl` was simplified to take `string` (not `string | undefined`) since `getOptionalEnv` always returns a string ("" when unset), and an empty input now naturally falls through the early-return path. While here, also added `http:`/`https:` scheme validation (per CodeRabbit's nearby suggestion) so a misconfigured `javascript:` / `data:` URL can't ever be embedded as a clickable link in alert HTML.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: server/api/src/services/notifier.ts [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+  const adminUrl = adminBase ? `${adminBase}/errors/${payload.apiErrorId}` : null;
+
+  const subject = `[zedi:${payload.severity}] API error ${payload.sentryIssueId}`;
```
[the remark this answers is not in the document]
Comment (otomatty) [reply server/api/src/services/notifier.ts]: Applied in 96da9b6 with a slightly tighter regex (`/[\r\n]+/g`) so consecutive CR/LF runs collapse into a single space:

```ts
const subject = `[zedi:${payload.severity}] API error ${payload.sentryIssueId}`.replace(
  /[\r\n]+/g,
  " ",
);
```

Resend takes the subject as a JSON field today so this isn't currently exploitable, but I agree this is cheap defense-in-depth for any future SMTP transport. Added a test (`件名から CR/LF を取り除く`) that asserts no CR/LF survives even when a malformed `sentryIssueId` is passed in.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: server/api/src/services/notifier.ts [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+ * @returns 送信結果 / Per-channel delivery result
+ */
+export async function notifyApiErrorAlert(
+  payload: NotifyApiErrorAlertPayload,
+): Promise<NotifyApiErrorAlertResult> {
+  if (!NOTIFIABLE_SEVERITIES.includes(payload.severity)) {
+    return { email: { sent: false } };
+  }
+
+  const to = process.env.MONITORING_NOTIFY_EMAIL?.trim();
```
[the remark this answers is not in the document]
Comment (otomatty) [reply server/api/src/services/notifier.ts]: Applied in 96da9b6:

```ts
const to = getOptionalEnv("MONITORING_NOTIFY_EMAIL").trim();
```

Note: `getOptionalEnv` returns `""` (not `undefined`) when the var is unset, so optional chaining (`?.trim()`) isn't needed — `"".trim()` is `""` and the subsequent `if (!to)` short-circuits to a no-op as before.

---
_Generated by [Claude Code](https://claude.ai/code)_
